### ISSUE TITLE
Problem making instance with Tripal JBrowse Management sub-module

### ISSUE THREAD

**Issue by btski** [tripal/tripal_jbrowse#20]
While attempting to make a new JBrowse instance through the Tripal JBrowse Management sub-module, I get the following error:

```
2019-05-07 16:29:07: Calling: tripal_jbrowse_mgmt_create_instance_files(2)
mkdir(): Permission denied tripal_jbrowse_mgmt_commands.inc:19       [warning]
Job execution failed: Unable to make data directory! Please make sure[error]
the directory 
      at  exists and is writable by the current user.
```

The error suggests that line 14 of the tripal_jbrowse/tripal_jbrowse_mgmt/includes/tripal_jbrowse_mgmt_commands.inc
file isn't finding the correct directory. When I try to launch a new JBrowse instance, I am redirected to my site home page (invalid path).

I have changed the permissions of the tripal_jbrowse directory
```
cd /var/www/html/sites/all/modules/tripal_jbrowse
sudo find . -type d -exec chmod 755 {} \; && sudo find . -type f -exec chmod 644 {} \;
sudo find . -type d -exec chown ubuntu:ubuntu {} \; && sudo find . -type f -exec chown ubuntu:ubuntu {} \;
```

Any other suggestions?

**Comment by almasaeed2010** [comment]
Hello @btski 

Have you configured the module by visiting `http://yoursite.org/admin/tripal/extension/tripal_jbrowse/management/configure`?

If so, can you share the paths you provided?

Thanks!

**Comment by btski** [comment]
Should the JBrowse dependencies have been installed with the module or is that a separate step? I ended up creating all of the recommended paths and edited permissions. Missing dependencies.

<img width="1337" alt="Screen Shot 2019-05-07 at 12 34 21 PM" src="https://user-images.githubusercontent.com/32686196/57320420-7e842080-70c4-11e9-9415-4b7d3ee40ca4.png">

Edit - manually installing JBrowse now.

**Comment by almasaeed2010** [comment]
Jbrowse should be installed independently. To do so, you simply need to run the following from within your `DRUPAL_ROOT/tools` folder. (if there isn't a tools folder, create one).

```bash
git clone https://github.com/GMOD/jbrowse.git
cd jbrowse
sh setup.sh
```

Note that you might have to install some perl dependencies. See https://github.com/GMOD/jbrowse for more info.

**Comment by bradfordcondon** [comment]
Perhaps the README should make more clear that this module requires a configured JBrowse instance as a prerequisite?

**Comment by almasaeed2010** [comment]
Yeah absolutely. I still haven't had the chance to move the documentation over o here. It's still on the old repo: https://hardwoods-jbrowse.readthedocs.io/en/latest/

**Comment by btski** [comment]
I don't believe the setup instructions on either the README or [online documentation](https://tripal_jbrowse.readthedocs.io) specified how to reach the configuration page. As far as I can tell, the page can only be reached by hyperlink. There is no visible config button via the Drupal modules menu.

**Comment by btski** [comment]
The Tripal JBrowse Management module configuration page lets you designate the full path of a data directory. This works reasonably well for creating instances, but is not recognized by the Tripal-JBrowse Integration module, which assumes you will put everything under a data directory within sites/tools/jbrowse. For example, my Tripal JBrowse Management configuration page currently looks like this:
<img width="725" alt="Screen Shot 2019-05-14 at 12 36 29 PM" src="https://user-images.githubusercontent.com/32686196/57719256-f3b3a080-7644-11e9-87e6-7b3d8b1fa4f3.png">

And I am getting the following error whenever I try to view a JBrowse instance:
<img width="968" alt="Screen Shot 2019-05-14 at 12 38 17 PM" src="https://user-images.githubusercontent.com/32686196/57719352-2e1d3d80-7645-11e9-933c-54a0771fcc82.png">

The directory sites/tools/jbrowse/data does not exist. Instead, it is located at sites/default/files/jbrowse, which is given in the Tripal JBrowse Management configuration page.

Lastly, I can't add tracks to an instance via the Tripal JBrowse Management module unless I first manually change the owner permissions of the data directory from ubuntu:ubuntu to ubuntu:www-data.

**Comment by almasaeed2010** [comment]
This configuration can only be done via server configuration (ie apache or nginx). To my knowledge, there is no way to tell JBrowse where the data folder is if it's not relative to the installation path. 

If you are using Apache, try adding the following to your .htaccess file (at the end of `<IfModule mod_headers.c>` block)

```
Alias /tools/jbrowse/data /var/www/html/sites/default/files/jbrowse
```

Let me know if that works and I'll add it to the docs.

Also, I noticed that your path to bin directory is not absolute. There is a good chance that this is required as indicated by the field description.

**Comment by btski** [comment]
Couldn't get alias to work, so I tried moving everything to /tools/jbrowse/data directory. After reconfiguring Tripal JBrowse Management, new instances are still being created in the old /var/www/html/sites/default/files/jbrowse directory.
<img width="713" alt="Screen Shot 2019-05-14 at 1 48 54 PM" src="https://user-images.githubusercontent.com/32686196/57723822-164ab700-764f-11e9-900c-2c160a9a04f7.png">

Uninstalling the Tripal JBrowse Management module did not correct the new data directory path. Post uninstallation, all Tripal JBrowse Management pages were wiped clean _except_ the configuration page, which kept all prior input settings. Note that everything seems to work once contained within the /tools/jbrowse/data directory. My current "fix" is to copy everything from my old directory to the new location. Any ideas why changes to the data directory path in the Tripal JBrowse Management configuration page are not being recognized?

Edit - Thought this might be helpful:
<img width="1008" alt="Screen Shot 2019-05-14 at 2 04 22 PM" src="https://user-images.githubusercontent.com/32686196/57725078-e05b0200-7651-11e9-9dbc-13f8fe022427.png">
Message requests that directory be deleted from new path. The request reflects the changes to the configuration page, but the file exists in the old path.

**Comment by almasaeed2010** [comment]
If you have access, try to add the alias in your apache config file. In ubintu it would be in `/etc/apache/sites-enabled/[SITE]`. In centos `/etc/httpd/conf.d/[your conf file]`

That's likely a bug. I'll look into it when I get a chance.

Thanks!

**Comment by laceysanderson** [comment]
Has any progress been made on this? @btski do you still need a solution?

**Comment by btski** [comment]
From memory, my solution was to just go with the default configurations because I couldn't get custom configurations to work. I left this issue open because it doesn't seem to be working as intended but close it if you can't reproduce the problem for yourself.

**Comment by laceysanderson** [comment]
Thanks for the update @btski! 

I tried the following:
1. Configure Tripal JBrowse Management with data directory within JBrowse directory as expected.
2. Created an instance
3. Changed the data directory configuration
4. Created a track for the instance -new config was used.
5. Created a new instance -new config was used.
6. **Checked the launch for both instances: `data/genus-species/data` was used.**

I've submitted a PR (#28) to fix the bug in 6 above as well as updated documentation to make it more clear that JBrowse needs to be installed before the module is used.

Regarding the Tripal JBrowse page integration module, it did assume that the data directory was within the JBrowse instance and was not configurable. As described in #26, we're moving away from the old Tripal JBrowse page integration module for better synergy between the two modules. These changes will also remove this assumption.
